testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java

```diff
@@ -442,6 +442,11 @@ protected void assertQueryFails(Session session, @Language("SQL") String sql, @L
         QueryAssertions.assertQueryFails(queryRunner, session, sql, expectedMessageRegExp);
     }
 
+    protected void assertQueryFailsIncorrectly(@Language("SQL") String sql, @Language("RegExp") String expectedMessageRegExp)
```

Comment:
There are several methods here added and removed two commits later so that they can be used eventually within 
the test `TestCorruptedIcebergTable`:

https://github.com/trinodb/trino/pull/16674/commits/5d2c4eb0692595974a8a24c3d56bbaad35aacf1f

Add those methods locally to `TestCorruptedIcebergTable`.